lib/elixir/src/elixir_parser.yrl

```diff
@@ -405,8 +406,16 @@ dot_op -> '.' eol : '$1'.
 dot_identifier -> identifier : '$1'.
 dot_identifier -> matched_expr dot_op identifier : build_dot('$2', '$1', '$3').
 
-dot_alias -> aliases : {'__aliases__', meta_from_token('$1', 0), ?exprs('$1')}.
-dot_alias -> matched_expr dot_op aliases : build_dot_alias('$2', '$1', '$3').
+dot_plain_alias -> aliases : {'__aliases__', meta_from_token('$1', 0), ?exprs('$1')}.
```

Comment:
I don't think we need to change the parser, I have found this old commit, we just need to bring it back:

https://github.com/elixir-lang/elixir/commit/b4b9ecccb4c03425bec62387da6628ff5b06504c

Except I wouldn't bring `[]`, although we can bring both `{}` and `<<>>`. The reason we shouldn't bring `[]` it is because we don't have `:[]` in AST anyway.